addons/im_livechat/static/src/embed/common/close_confirmation.js

```diff
@@ -0,0 +1,38 @@
+import { Component, useRef, useEffect } from "@odoo/owl";
+
+export class CloseConfirmation extends Component {
+    static template = "im_livechat.CloseConfirmation";
+    static props = ["onCloseConfirmationDialog", "onClickLeaveConversation"];
+
+    setup() {
+        this.confirmationDialogRef = useRef("closeConfirmation");
+        useEffect(
+            () => {
+                if (this.confirmationDialogRef.el) {
+                    this.confirmationDialogRef.el.focus();
+                }
+            },
+            () => [this.confirmationDialogRef.el]
+        );
+    }
+
+    onKeydown(ev) {
+        if (ev.key === "Escape") {
+            this.props.onCloseConfirmationDialog();
+        } else if (ev.key === "Enter") {
+            this.props.onClickLeaveConversation();
+        }
+    }
+
+    onClick(ev) {
+        const targetClass = ev.target.classList;
+        if (
+            targetClass.contains("o-livechat-CloseConfirmation") ||
+            targetClass.contains("btn-close")
+        ) {
+            this.props.onCloseConfirmationDialog();
+        } else if (targetClass.contains("o-livechat-CloseConfirmation-leave")) {
+            this.props.onClickLeaveConversation();
+        }
+    }
```

Comment:
Better `t-on-click` immediately on the relevant target. Global click and conditional here requires to maintain these LOCs with the template, whereas `t-on-` on immediate nodes require no effort